### ISSUE TITLE
Manually count CC for BoundSet::deriveImpliedBounds

### ISSUE THREAD

**Issue by Perpolo** [DD2480-Group13/javaparser#5]
by program - 33

by me:
if - 20
&& - 9
return - 3
throw - 1

total:
20+9-3-1+2=27

**Comment by fnosse** [comment]
I don't think you can count the returns as exit points, since they are only returning within the lambda expression.

I got 29. 

So I had to count by drawing graphs for each "segment" and then connecting them by edges. Ex. The first "segment" newContraintsSet-assignment has 16 edges and 13 nodes, resulting in +3 complexity. 

So in the end the "segments" were 3, 1, 1, 0, 3, 1, 4, 1, 4. Sum = 18, +9 edges because there are ten "segments" to connect. M = 18 + 9 + 2 = 29.

After discussion I agree with 27, probably did something wrong when drawing graphs.
